crucible-mir/src/Mir/Trans.hs

```diff
@@ -1973,14 +1974,41 @@ genFn :: HasCallStack =>
     C.TypeRepr ret ->
     Ctx.Assignment (R.Atom s) args ->
     MirGenerator h s ret (G.Label s)
-genFn (M.Fn fname argvars _ body@(MirBody localvars blocks _)) rettype inputs = do
+genFn (M.Fn fname argvars sig body@(MirBody localvars blocks _)) rettype inputs = do
 
   lm <- buildLabelMap body
   labelMap .= lm
 
   let addrTaken = mconcat (map addrTakenVars blocks)
   initLocals (argvars ++ localvars) addrTaken
-  initArgs inputs (reverse argvars)
+  let inputExps = toListFC (\atom -> MirExp (R.typeOfAtom atom) (R.AtomExpr atom)) inputs
+  inputExps' <- case sig ^. M.fsabi of
+    M.RustCall (M.RcSpreadArg spreadArg) -> do
+      -- The Rust signature looks like `(T, (U, V))`, but the Crucible version
+      -- uses the untupled form `(T, U, V)`.  `argvars` follows the tupled
+      -- form, but `inputExps` follows the untupled form.  Here we convert
+      -- `inputExps` to match `argvars` by tupling up some of the arguments.
+      --
+      -- Arguments from `spreadArg` onward need to be tupled.  Note that
+      -- `fsspreadarg` is 1-based (it's a MIR `Local` ID, and the first
```

Comment:
Do you mean `spreadArg` here, instead of `fsspreadarg`?

Reply:
Fixed in 35f00b44baee28f5649bba773df875c95d3b71be